src/sshfs_mount/sftp_server.cpp

```diff
@@ -493,17 +540,16 @@ int mp::SftpServer::handle_mkdir(sftp_client_message msg)
         return reply_failure(msg);
     }
 
-    QFileInfo current_dir(filename);
-    QFileInfo parent_dir(current_dir.path());
-    int rev_uid = reverse_uid_for(msg->attr->uid, parent_dir.ownerId());
-    int rev_gid = reverse_gid_for(msg->attr->gid, parent_dir.groupId());
+    int rev_uid = reverse_uid_for(parent_dir.ownerId(), parent_dir.ownerId());
+    int rev_gid = reverse_gid_for(parent_dir.groupId(), parent_dir.groupId());
```

Comment:
This was originally:
```
int rev_uid = reverse_uid_for(msg->attr->uid, parent_dir.ownerId());
int rev_gid = reverse_gid_for(msg->attr->gid, parent_dir.groupId());
```
but, libssh doesn't populate `msg->attr->uid` or the gid. Is there some scenario that I'm not thinking of where this does get populated? The same was also being done in `handle_open()`...